pkg/server/config.go

```diff
@@ -45,87 +50,107 @@ func DefaultConfig() *Config {
 		ResourcesToSync:          []string{"deployments.apps"},
 		RootDirectory:            ".kcp",
 		SyncerImage:              "quay.io/kcp-dev/kcp-syncer",
+		Authentication:           kubeoptions.NewBuiltInAuthenticationOptions().WithAll(),
 	}
 }
 
 // ConfigFromFlags returns a default configuration with config values set from the provided flag set
 func ConfigFromFlags(flags *pflag.FlagSet) *Config {
 	cfg := DefaultConfig()
-	listen, err := flags.GetString("listen")
-	if err == nil {
+
+	if listen, err := flags.GetString("listen"); err == nil {
```

Comment:
This code gets removed in #228 so I wouldn't try *too* hard to refactor if you have anything else planned ;)

Reply:
Excellent. I'll wait for that then. This was all pretty annoying because this seemed to want to be set up using [`AddFlags`](https://github.com/kubernetes/kubernetes/blob/63e7ee43bb5394abd6dd5010be47ebcb22ec0ef8/pkg/kubeapiserver/options/authentication.go#L245) but the way we added and parsed flags as two separate steps made that harder. I bet your change will make this all easier 👍